level_5/lexer.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include <iostream>
+#include <stdexcept>
+#include <variant>
+#include <cctype>
+
+enum class Type {
+	INTEGER, PLUS, MINUS, MUL, DIV, END, LPARAN, RPARAN
+};
+
+class Token{
+	private:
+	Type token_type_;
+	std::variant<int, char> value_;
+	public:
+		Token(Type token_type, int value) : token_type_{token_type}, value_{value} {}
```

Comment:
Also provide a constructor for non integer types, such as operators, which do not need that second parameter.